### ISSUE TITLE
Terraform crash while reading azure app service certificate

### ISSUE THREAD

**Issue by happyfeet23** [hashicorp/terraform-provider-azurerm#13372]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Terraform (and AzureRM Provider) Version

* Terraform Core version: 0.14.3
* AzureRM Provider version: 2.72.0

### Terraform Configuration Files

```hcl
resource "azurerm_app_service_custom_hostname_binding" "domain_binding" {
  hostname            = var.environment == "prod" ? var.domain : "${var.environment}.${var.domain}"
  app_service_name    = "${var.project_short}-${var.environment}-app"
  resource_group_name = module.az_webapp.resource_group_name
}

resource "azurerm_app_service_managed_certificate" "domain_cert" {
  custom_hostname_binding_id = azurerm_app_service_custom_hostname_binding.domain_binding.id
}

resource "azurerm_app_service_certificate_binding" "domain_cert_binding" {
  hostname_binding_id = azurerm_app_service_custom_hostname_binding.domain_binding.id
  certificate_id      = azurerm_app_service_managed_certificate.domain_cert.id
  ssl_state           = "SniEnabled"
}
```

### Description / Feedback

It should create a certificate for the hostname and then the binding. But terraform crashes after creating the certificate. The state doesnt receive the certificate created. 

The crash output is as below:
11:20:23  azurerm_app_service_managed_certificate.domain_cert: Still creating... [1m0s elapsed]
11:20:27  
11:20:27  Error: rpc error: code = Unavailable desc = transport is closing
11:20:27  
11:20:27  
11:20:27  
11:20:27  Error: authorization.RoleAssignmentsClient#Create: Failure responding to request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="RoleAssignmentExists" Message="The role assignment already exists."
11:20:27  
11:20:27    on roleassignments.tf line 8, in resource "azurerm_role_assignment" "storage_appservice_queueCon":
11:20:27     8: resource "azurerm_role_assignment" "storage_appservice_queueCon" {
11:20:27  
11:20:27  
11:20:27  
11:20:27  Error: authorization.RoleAssignmentsClient#Create: Failure responding to request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="RoleAssignmentExists" Message="The role assignment already exists."
11:20:27  
11:20:27    on roleassignments.tf line 15, in resource "azurerm_role_assignment" "storage_appservice_blobCon":
11:20:27    15: resource "azurerm_role_assignment" "storage_appservice_blobCon" {
11:20:27  
11:20:27  
11:20:27  
11:20:27  Error: authorization.RoleAssignmentsClient#Create: Failure responding to request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="RoleAssignmentExists" Message="The role assignment already exists."
11:20:27  
11:20:27    on roleassignments.tf line 30, in resource "azurerm_role_assignment" "sql_dbbackup_contributor_dev":
11:20:27    30: resource "azurerm_role_assignment" "sql_dbbackup_contributor_dev" {
11:20:27  
11:20:27  
11:20:27  panic: runtime error: invalid memory address or nil pointer dereference
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x540d82a]
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: goroutine 107 [running]:
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: github.com/hashicorp/terraform-provider-azurerm/internal/services/web.resourceAppServiceManagedCertificateRead(0xc001359280, 0x580f2c0, 0xc000c86380, 0x0, 0x0)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/internal/services/web/app_service_managed_certificate_resource.go:221 +0x6ea
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: github.com/hashicorp/terraform-provider-azurerm/internal/services/web.resourceAppServiceManagedCertificateCreateUpdate(0xc001359280, 0x580f2c0, 0xc000c86380, 0x0, 0x0)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/internal/services/web/app_service_managed_certificate_resource.go:192 +0xd5d
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000c728c0, 0x67b0388, 0xc00133a780, 0xc001359280, 0x580f2c0, 0xc000c86380, 0x0, 0x0, 0x0)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:318 +0x1ee
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000c728c0, 0x67b0388, 0xc00133a780, 0xc0016585b0, 0xc0017d2a00, 0x580f2c0, 0xc000c86380, 0x0, 0x0, 0x0, ...)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:456 +0x67b
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000ad2318, 0x67b0388, 0xc00133a780, 0xc0014d4ff0, 0xc00133a780, 0x5cf39c0, 0xc001adb900)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:955 +0x8ef
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc00137bd60, 0x67b0430, 0xc00133a780, 0xc001658460, 0xc00137bd60, 0xc001adb980, 0xc001444ba0)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:332 +0xb5
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x5cf39c0, 0xc00137bd60, 0x67b0430, 0xc001adb980, 0xc000b46300, 0x0, 0x67b0430, 0xc001adb980, 0xc0008cc380, 0x376)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000992380, 0x67fd1b8, 0xc000103800, 0xc000f95560, 0xc00087c990, 0x9d78120, 0x0, 0x0, 0x0)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc000992380, 0x67fd1b8, 0xc000103800, 0xc000f95560, 0x0)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000bd41c0, 0xc000992380, 0x67fd1b8, 0xc000103800, 0xc000f95560)
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
11:20:27  2021-09-16T05:50:26.428Z [DEBUG] plugin.terraform-provider-azurerm_v2.72.0_x5: 	/opt/teamcity-agent/work/blahblahblah/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd
11:20:27  2021-09-16T05:50:26.434Z [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
11:20:27  2021/09/16 05:50:26 [DEBUG] azurerm_app_service_managed_certificate.domain_cert: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
11:20:27  2021-09-16T05:50:26.434Z [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/azurerm/2.72.0/linux_amd64/terraform-provider-azurerm_v2.72.0_x5 pid=1325 error="exit status 2"
11:20:27  2021/09/16 05:50:26 [TRACE] EvalMaybeTainted: azurerm_app_service_managed_certificate.domain_cert encountered an error during creation, so it is now marked as tainted
11:20:27  2021/09/16 05:50:26 [TRACE] EvalWriteState: removing state object for azurerm_app_service_managed_certificate.domain_cert
11:20:27  2021/09/16 05:50:26 [TRACE] EvalApplyProvisioners: azurerm_app_service_managed_certificate.domain_cert has no state, so skipping provisioners
11:20:27  2021/09/16 05:50:26 [TRACE] EvalMaybeTainted: azurerm_app_service_managed_certificate.domain_cert encountered an error during creation, so it is now marked as tainted
11:20:27  2021/09/16 05:50:26 [TRACE] EvalWriteState: removing state object for azurerm_app_service_managed_certificate.domain_cert
11:20:27  2021/09/16 05:50:26 [TRACE] vertex "azurerm_app_service_managed_certificate.domain_cert": visit complete
11:20:27  2021/09/16 05:50:26 [TRACE] dag/walk: upstream of "azurerm_app_service_certificate_binding.domain_cert_binding" errored, so skipping
11:20:27  2021/09/16 05:50:26 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
11:20:27  2021/09/16 05:50:26 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/azurerm\"] (close)" errored, so skipping
11:20:27  2021/09/16 05:50:26 [TRACE] dag/walk: upstream of "root" errored, so skipping
11:20:27  2021/09/16 05:50:26 [DEBUG] Uploading remote state to S3: {
11:20:27    Body: buffer(0xc0010f7620),
11:20:27    Bucket: "blahblah.tfstate-bucket",
11:20:27    ContentLength: 260543,
11:20:27    ContentType: "application/json",
11:20:27    Key: "environment/dev/terraform.tfstate",
11:20:27    ServerSideEncryption: "AES256"
11:20:27  }
11:20:27  2021-09-16T05:50:26.558Z [DEBUG] plugin: plugin exited
11:20:27  
11:20:27  
11:20:27  
11:20:27  !!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
11:20:27  
11:20:27  Terraform crashed! This is always indicative of a bug within Terraform.
11:20:27  A crash log has been placed at "crash.log" relative to your current
11:20:27  working directory. It would be immensely helpful if you could please
11:20:27  report the crash with Terraform[1] so that we can fix this.
11:20:27  
11:20:27  When reporting bugs, please include your terraform version. That
11:20:27  information is available on the first line of crash.log. You can also
11:20:27  get it by running 'terraform --version' on the command line.
11:20:27  
11:20:27  SECURITY WARNING: the "crash.log" file that was created may contain 
11:20:27  sensitive information that must be redacted before it is safe to share 
11:20:27  on the issue tracker.
11:20:27  
11:20:27  [1]: https://github.com/hashicorp/terraform/issues
11:20:27  
11:20:27  !!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by protoblade** [comment]
welcome to the club [13357](https://github.com/hashicorp/terraform-provider-azurerm/pull/13357). its already merged into 2.77 but its not released yet

**Comment by jackofallops** [comment]
Closed by #13357
